src/TTrigger.cpp

```diff
@@ -291,13 +291,13 @@ bool TTrigger::match_perl(char* subject, const QString& toMatch, int regexNumber
     int namecount;
     int name_entry_size;
 
-    int subject_length = strlen(subject);
+    int subject_length = toMatch.size();
     int rc, i;
     std::list<std::string> captureList;
     std::list<int> posList;
     int ovector[300]; // 100 capture groups max (can be increase nbGroups=1/3 ovector
 
-    rc = pcre_exec(re.data(), nullptr, subject, subject_length, 0, 0, ovector, 100);
+    rc = pcre16_exec(re.data(), nullptr, toMatch.utf16(), subject_length, 0, 0, ovector, 100);
```

Comment:
I'm not totally sure about this - and it was so long ago that I played with this that I cannot remember the PCRE/PCRE_16 APIs but I am suspicious that the 3rd argument has changed from the *`subject`* to the *`toMatch`* argument...

Reply:
_toMatch_ is QString equivalent of _subject_, look at _TriggerUnit::processDataStream_. From there _match_ arguments are passed unchanged to _match_perl_. It seemed easier work with original QString than to try to convert _subject_ back to utf.

Reply:
Fair enough.